pkg/ingester/ingester_v2_test.go

```diff
@@ -2422,3 +2422,39 @@ func TestIngesterNotDeleteUnshippedBlocks(t *testing.T) {
 		require.NotEqual(t, b.Meta().ULID, newBlocks2[1].Meta().ULID)
 	}
 }
+
+func TestIngesterPushErrorDuringForcedCompaction(t *testing.T) {
+	i, cleanup, err := newIngesterMockWithTSDBStorage(defaultIngesterTestConfig(), nil)
+	require.NoError(t, err)
+	t.Cleanup(cleanup)
```

Comment:
I think ideally it would be good to also add a test that verifies compactHead waits on no pushes to be in flight and locks the muutex.

Reply:
I tried figuring out possible ways to test this, but since it uses a `sync.WaitGroup` for the in-flight requests and all the operations are relatively quick, I am not sure how to deterministically test for no in-flight requests. Do you have any ideas?

Reply:
One possibility is to call `acquireAppendLock` directly from the test, then call compaction, verify that it updated bool hasn't started compaction, and then call `releaseAppendLock` to simulate finished push.

Verification of compaction could be done via TSDB metrics, or Head minT.

Reply:
Nice one, I have added a test with that logic.